lib/js/src/manager/screen/utils/SoftButtonObject.js

```diff
@@ -239,6 +243,157 @@ class SoftButtonObject {
+
+        if (this.getCurrentState().getSystemAction() !== null && this.getCurrentState().getSystemAction() !== undefined) {
+            currentState.setSystemAction(JSON.parse(JSON.stringify(this.getCurrentState().getSystemAction())));
+        }
+        const listenerClone = (this.getOnEventListener() !== undefined && this.getOnEventListener() !== null) ? this.getOnEventListener().bind({}) : null;
+        const clone = new SoftButtonObject(
+            JSON.parse(JSON.stringify(this.getName())),
+            statesClone,
+            currentState.getName(),
+            listenerClone
+        );
+        if (this._updateListener !== null && this._updateListener !== undefined) {
+            clone._setUpdateListener(this._updateListener.bind(clone));
+        }
+
+        clone._buttonId = JSON.parse(JSON.stringify(this.getButtonId()));
+        return clone;
+    }
+
+    /**
+     * Checks whether two SoftButtonObjects can be considered equivalent, but does NOT compare the listener objects
+     * @param {SoftButtonObject} other - The object to compare
+     * @returns {Boolean} - Whether the objects are the same or not
+     */
+    equals (other) {
+        if (other === null || other === undefined) {
+            return false;
+        }
+        if (this === other) {
+            return true;
+        }
+        if (!(other instanceof SoftButtonObject)) {
+            return false;
+        }
+        if (this.getName() !== other.getName()) {
+            return false;
+        }
+        const states = this.getStates();
+        const otherStates = other.getStates();
+
+        if ((states !== null && otherStates === null) || (states === null && otherStates !== null)) {
+            return false;
+        }
+        if ((states !== null && otherStates !== null)) {
+            if (states.length !== otherStates.length) {
+                return false;
+            }
+            for (let index = 0; index < states.length; index++) {
+                if (states[index] === null && otherStates[index] === null) {
+                    continue;
+                }
+                if (states[index].getName() !== otherStates[index].getName()) {
+                    return false;
+                }
+                if (states[index].getHighlighted() !== otherStates[index].getHighlighted()) {
+                    return false;
+                }
+                if (states[index].getSystemAction() !== otherStates[index].getSystemAction()) {
+                    return false;
+                }
+                if (states[index].getSoftButton().getType() !== otherStates[index].getSoftButton().getType()) {
+                    return false;
+                }
+                if (states[index].getSoftButton().getText() !== otherStates[index].getSoftButton().getText()) {
+                    return false;
+                }
+                if (states[index].getSoftButton().getImage() !== otherStates[index].getSoftButton().getImage()) {
```

Comment:
This getImage() comparison will need a deep object equality check too

Reply:
Added some extra checks to compare for deep equality.